amical/data_processing.py

```diff
@@ -117,13 +117,22 @@ def select_data(cube, clip_fact=0.5, clip=False, verbose=True, display=True):
         ind_clip = []
         cube_cleaned_checked = np.array(good_fram)
 
+    cube_cleaned_checked = np.array(cube_cleaned_checked)
+
     ind_clip2 = np.where(fluxes <= limit_flux)[0]
     if ((worst_fr in ind_clip2) and clip) or (worst_fr in flag_fram):
         ext = "(rejected)"
     else:
         ext = ""
 
-    diffmm = 100 * abs(np.max(fluxes) - np.min(fluxes)) / med_flux
+    fluxes_check = np.array([x.sum() for x in cube_cleaned_checked])
+
+    best_fr = np.argmax(fluxes_check)
+    worst_fr = np.argmin(fluxes_check)
+
+    med_flux = np.median(fluxes_check)
+    std_flux = np.std(fluxes_check)
+    diffmm = 100 * abs(np.max(fluxes_check) - np.min(fluxes_check)) / med_flux
```

Comment:
`np.ptp(x)` does the same as `np.max(x) - np.min(x)`, but in a single data pass
```suggestion
    diffmm = 100 * abs(np.ptp(fluxes_check)) / med_flux
```